alacritty/src/event.rs

```diff
@@ -393,9 +404,6 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         self.search_state.origin = if self.terminal.mode().contains(TermMode::VI) {
             self.terminal.vi_mode_cursor.point
         } else {
-            // Clear search, since it is used as the active match.
-            self.terminal.selection = None;
```

Comment:
I think we should clear selection, so it won't obscure your new focused match? Though we shouldn't do that for Vi mode, since you can move selection via Vi cursor and it's intentional most of the time that way.

Reply:
I'm not sure what you mean?

Reply:
Start normal search, type something, leave normal search, start search again, starting typing the same request, you should have focused match obscured by selection.

Reply:
Hmmm, I do now understand your issue, but I'm not sure if we want to fix that or not. Generally it's trivial either way, but I'm not sure what's the better behavior here. I'll follow up after work today.